### PR TITLE
Fix getURI() with empty uri query.

### PR DESCRIPTION
**Pull request by boaks**
Add unit test for empty uri path and empty uri query.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
If no objections, I will go to merge this next week.

File: californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java [hunk 1]
```diff
@@ -401,11 +404,14 @@ public String getURI() {
...
-		String path = new StringBuilder("/").append(getOptions().getUriPathString()).toString();
-		String query = getOptions().getUriQueryString();
+		// according RFC7252, section 6.5, item 7, a empty resource name is represented by "/" as path. 
+		// therefore always use the leading "/", even if the uri path is empty. 
+		String path = "/" + getOptions().getUriPathString();
+		String query = getOptions().getURIQueryCount() > 0 ? getOptions().getUriQueryString() : null;
 		try {
 			URI uri = new URI(getScheme(), null, host, port, path, query, null);
-			return uri.toASCIIString();
+			// ensure, that none-ascii characters are "percent-encoded"
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java]: should read `... non-ascii characters ....`

File: californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java [hunk 2]
```diff
@@ -401,11 +404,14 @@ public String getURI() {
 		} else {
 			port = -1;
 		}
-		String path = new StringBuilder("/").append(getOptions().getUriPathString()).toString();
-		String query = getOptions().getUriQueryString();
+		// according RFC7252, section 6.5, item 7, a empty resource name is represented by "/" as path. 
+		// therefore always use the leading "/", even if the uri path is empty. 
+		String path = "/" + getOptions().getUriPathString();
+		String query = getOptions().getURIQueryCount() > 0 ? getOptions().getUriQueryString() : null;
```
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java]: assuming that this is about preventing to get back just a `?`, wouldn't it make sense to include the check in `Options.getUriQueryString()` itself?

Comment (boaks) [inline californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java]: The other Option.getXyZString() returns also "" instead of null, so I decided to do this in Request.getURI().

**Comment by boaks** [comment]
Typo corrected.

File: californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java (hunk 2 above)
Comment (sophokles73) [inline californium-core/src/main/java/org/eclipse/californium/core/coap/Request.java]: I see

**Comment by sophokles73** [comment]
LGTM
